examples/all/manifests/thanos-compactor-deployment.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
Why not statefulset? It's nice to have some persistent volume... We might use some disk caching on upgrade/restart. It's not super important now because when I looked at the current code we download meta files anyway :man_shrugging: but we should definitely has some smarter sync code and there is some work to unify and reuse disk for caching of those. 

So moving to statefulset straight away would be nice TBH unless it's major pain to set up.